go.mod

```diff
@@ -1,5 +1,6 @@
 module github.com/cilium/cilium-cli
 
+// renovate: datasource=golang-version depName=go
 go 1.21.1
```

Comment:
We may not want renovate to bump the patch version in the go.mod, see https://github.com/cilium/cilium/pull/28686

Reply:
oh interesting could we just specify `go 1.21` here?

Reply:
anyways i'll just wait for https://github.com/cilium/cilium/pull/28686 to get merged and do the same thing. converting to draft.

Reply:
I think that you can specify `go 1.21` as long as you add a toolchain directive. Maybe that's what we should be doing, ie:
```
go 1.21

// renovate: datasource=golang-version depName=go
toolchain 1.21.3
```

Reply:
fascinating

Reply:
cc @joestringer :point_up: